src/Core/Content/Seo/SalesChannel/StoreApiSeoResolver.php

```diff
@@ -60,14 +60,24 @@ public function addSeoInformation(ResponseEvent $event): void
             return;
         }
 
-        if (!$event->getRequest()->headers->has(PlatformRequest::HEADER_INCLUDE_SEO_URLS)) {
+        $request = $event->getRequest();
+
+        if (!$request->headers->has(PlatformRequest::HEADER_INCLUDE_SEO_URLS)) {
+            return;
+        }
+
+        /** @var SalesChannelContext|null $context */
+        $context = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_CONTEXT_OBJECT);
+
+        // This is likely the case for routes with the `auth_required` option set to `false`.
+        if (null === $context) {
```

Comment:
instead of checking for the object being `null`, you should check with `instanceof` for the type you are expecting. This would also make the "var" annotation in line 69 obsolete.